app/controllers/master/api/providers_controller.rb

```diff
@@ -157,8 +159,17 @@ def provider_account
     @provider_account ||= current_account.providers.without_deleted(!action_includes_deleted_providers?).find(params[:id])
   end
 
+  def signup_result_with_nil_token
+    signup_result = Signup::ResultWithAccessToken.new(account: provider_account, user: provider_account.admin_users.first)
+
+    # Signup::ResultWithAccessToken because it is the representer, but after it initializes, it builds an access token,
+    # which is a known design problem). So this access token needs to be set to nil before responding because it is an unsaved one
+    signup_result.access_token = nil
+    signup_result
+  end
+
   def action_includes_deleted_providers?
-    %w[show update].include?(action_name)
+    %w[show update plan_upgrade].include?(action_name)
```

Comment:
This means we allow the action for deleted providers? If so, why?

Reply:
Yeah, you're right, I guess it doesn't make sense to upgrade the plan of the provider that is scheduled for the deletion. I'll remove this.